### PR TITLE
Application type e2e definition

### PR DESCRIPTION
**Pull request by foxish**
Adding a new test for the application type in https://github.com/kubernetes-sigs/application

cc @krzyzacy @kow3ns

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -5848,6 +5848,36 @@ postsubmits:
...
+  name: application-periodic-default-gke
+  labels:
+    preset-service-account: true
+    preset-k8s-ssh: true
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20180418-79a97ca02-master
+      args:
+      - "--job=$(JOB_NAME)"
+      - "--repo=github.com/foxish/application=master"
```
Comment (krzyzacy) [inline prow/config.yaml]: your own fork for now? xD

Comment (foxish) [reply prow/config.yaml]: Yeah, will change it once the testing is stable. Thanks for the review!

**Comment by foxish** [comment]
Will address @BenTheElder's comments in a later PR along with pointing from my repository to the official one in kubernetes-sigs.

File: jobs/config.json [hunk 2]
```diff
@@ -1,4 +1,24 @@
 {
+  "application-periodic-default-gke": {
+    "args": [
+      "--check-leaked-resources",
+      "--cluster=",
```
[the remark this answers is not in the document]
Comment (krzyzacy) [reply jobs/config.json]: https://github.com/kubernetes/test-infra/blob/master/scenarios/kubernetes_e2e.py#L609, not the case here

**Pull request merged**